src/env/commands/emufs.S

```diff
@@ -47,206 +47,122 @@
 Header:
-	.long	-1		# link to next device driver
-	.word	0		# attribute word for driver
-	.word	Strat		# ptr to strategy routine
-	.word	Intr		# ptr to interrupt service routine
-	.ascii	"LINUX   "	# logical-device name
-
-# the Strat and Intr routines are entered with a "far call".  I don't
-# know how to tell as86 that asI would in Turbo Assembler, so Ijust
-# explicitly "retf" at the end of their execution.  careful!
-
-				# BPB from horse.mach.cs.cmu.edu
-Bpb:	.word	0x200		# Bytes per Sector
-	.byte	4		# Sectors per allocation unit
-	.word	1		# Number of reserved sectors
-	.byte	2		# Number of FATS
-	.word	0x200		# Max number of root entries
-	.word	0xac21		# Total number of sectors
-	.byte	0xf8		# Media Desc Byte
-	.word	0x2b		# Num sec per FAT
-	.word	0x23		# Sectors per track
-	.word	0xf		# Number of heads
-	.word	0		# Number of hidden sectors
-	.word	0		# High order word of hidden sectors
-	.long	0		# zero
-	.long	0
-	.long	0
-	.long	0
-	.long	0
-	.long	0
-	.long	0
-	.long	0
-
-BPBptr:	.word	Bpb
-
-RHPtr:	.long	0		# ptr to request header
-
-InitDone: .word 0		# 1 when initialisation is complete
-major:	.byte	0
-minor:	.byte	0
-sdasize: .word  0
+	.long	-1			# link to next device driver
+	.word   0b1000000000000000	# attribute word for char driver
+	.word	Strat			# ptr to strategy routine
+	.word	Intr			# ptr to interrupt service routine
+	.ascii	"EMUFS$  "		# logical-device name
 
-Dispatch:
+.align 2
 
-	.word	Init		# initialize driver
-	.word	MediaChk
-	.word	BuildBpb
+RHPtr:	.long	0			# ptr to request header
+
+Dispatch:
+	.word	Init			# initialize driver
 	.word	Dummy
-	.word	Read		# read
 	.word	Dummy
 	.word	Dummy
+	.word	Dummy			# read
 	.word	Dummy
-	.word	Write		# write
-	.word	Write		# write with verify
+	.word	Dummy
+	.word	Dummy
+	.word	Dummy			# write
+	.word	Dummy			# write with verify
 	.word	Dummy
 	.word	Dummy
 	.word	Dummy
 
 Strat:
-# was...
-#	mov	word ptr cs:[RHPtr],bx
-#	mov	word ptr cs:[RHPtr+2],es
 	movw	%bx, %cs:RHPtr
 	movw	%es, %cs:RHPtr+2
 	lret
 
-
 Intr:
-	pushw	%ax
-	pushw	%bx
-	pushw	%cx
-	pushw	%dx
+	pusha
 	pushw	%ds
 	pushw	%es
-	pushw	%di
-	pushw	%si
-	pushw	%bp
-	
+
 	pushw	%cs
 	popw	%ds
 
-	les	RHPtr,%di	# let es:di = request header
+	les	RHPtr,%di		# let es:di = request header
 
-	movb	%es:2(%di),%bl
+	movb	%es:CMD(%di),%bl
 	xorb	%bh,%bh
 	cmpw	$MaxCmd,%bx
-	jle	Intr1
-	call	Error
-	jmp	Intr2
+	jle	1f
+	movw	$0x8003,%ax		# Set error
+	jmp	2f
+
+1:	shlw	%bx
 
-Intr1:	shlw	%bx
-	
 	call	*Dispatch(%bx)
 
 	les	RHPtr,%di
 
-Intr2:	or	$0x100,%ax		# Merge done bit with status
-	movw	%ax,%es:3(%di)
+2:	or	$0x100,%ax		# Merge done bit with status
+	movw	%ax,%es:STATUS(%di)
 
-	popw	%bp
-	popw	%si
-	popw	%di
 	popw	%es
 	popw	%ds
-	popw	%dx
-	popw	%cx
-	popw	%bx
-	popw	%ax
+	popa
 	lret
 
 Dummy:
-	call MFSini
-	les	RHPtr, %di
-	movb	%es:2(%di),%bl
-	movb	$3, %bh
+	push	%es
+	push	%dx
 
-	movw    $DOS_HELPER_MFS_HELPER, %ax
+	push	%cs
+	pop	%es
+	movw	$.LDummyMsg, %dx
+	movw	$DOS_HELPER_PRINT_STRING, %ax
```

Comment:
It prints to the dosemu log file that DOS called the Dummy() function for some reason. I could remove it if necessary, but I don't think it should be printed in DOS as we always return 'no error' to DOS.